connman/src/storage.c

```diff
@@ -794,11 +796,60 @@ static void keyfile_cleanup(void)
 	keyfile_hash = NULL;
 }
 
+static int validate_path(const char *pathname)
+{
+	char resolved_path[PATH_MAX] = { 0 };
+
+	errno = 0;
+
+	/*
+	 * realpath() expands all symbolic links and references. Returns
+	 * ENOENT if pathname is NULL.
+	 */
+	if (!realpath(pathname, resolved_path)) {
+		/* File does not exist when saving, return 0.*/
+		if (errno == ENOENT)
```

Comment:
This isn't quite enough, you should still check if resolved_path matches pathname in this case, otherwise this can still be used to create new configs in another user's dirs with symlinks.

Reply:
Yes when this now uses `resolved_path`, added a commit 4b2bc8804c946fdaa4feb0b756859db39cbe726b to address this and cleaned up some bits.